tests/test_outlier.py

```diff
@@ -498,9 +498,18 @@ def test_ood_predictions_scores():
         adjust_pred_probs=False,
     )
 
+    ood_predictions_scores_gen, _ = outlier._get_ood_predictions_scores(
+        pred_probs=pred_probs,
+        method='gen',
+        adjust_pred_probs=False,
+        M=3, # Totally three classes
+    )
+
     # check OOD scores calculated correctly
     assert (1.0 - get_normalized_entropy(pred_probs) == ood_predictions_scores_entropy).all()
     assert (pred_probs.max(axis=1) == ood_predictions_scores_least_confidence).all()
+    assert ood_predictions_scores_gen.max() < 1 
+    assert ood_predictions_scores_gen.min() > 0
 
```

Comment:
@coding-famer  Can you confirm based on your comment that setting `X_ood` = mean of all classes' feature distributions works well?  I.e. then `X_ood` gets one of the lowest outlier scores amongst the entire dataset?

If so, could you update this unit test to use that value of `X_ood` and then also ensure the resulting outlier score for `X_ood` is very low (say within bottom 1-5% of scores across the whole dataset) ?

I think that is the final thing I'd like to see for this PR and then it's good to go!

Reply:
For `least_confidence` and `gen` it will be the second smallest and for `entropy` it will be the third smallest. 
So maybe I can add assert like `assert np.where(np.sort(ood_predictions_scores_entropy) == ood_predictions_scores_entropy[-1])[0] < 0.02 * len(ood_predictions_scores_entropy)` to ensure it will be within the bottom 2%?
What's more, do you think sampling several ood datapoints from another distribution instead of selecting one single value will be a better choice?

Reply:
I think we can just stick with a single `X_ood` point to keep the unit test simpler. Your plan sounds good!